network/peer.go

```diff
@@ -836,3 +879,45 @@ func (p *peer) getIP() utils.IPDesc {
 	defer p.ipLock.RUnlock()
 	return p.ip
 }
+
+// addAlias marks that we have found another
+// IP that we can connect to this peer at.
+//
+// assumes stateLock is held
+func (p *peer) addAlias(ip utils.IPDesc) {
+	p.ipLock.Lock()
```

Comment:
This doesn't seem to be needed. You are not accessing peer.ip..  And f you do you should use getIP()

Reply:
I utilize `ipLock` as a guard for `aliases` as well. Open to adding another lock if you prefer. Note: https://github.com/ava-labs/avalanchego/pull/746/files#diff-b06a14805cb9806d82ef18d987b32228ab9b36a4e679c7c128efd711038991b2R69

Reply:
ok, sorry I missed that.